lmnet/lmnet/blocks.py

```diff
@@ -126,3 +128,192 @@ def lmnet_block(
+
+    with tf.variable_scope(name):
+
+        output_1x1 = conv_bn_act(
+            "bottleneck_1x1",
+            inputs,
+            filters=bottleneck_channel,
+            kernel_size=1,
+            weight_decay_rate=weight_decay_rate,
+            is_training=is_training,
+            activation=activation,
+            batch_norm_decay=batch_norm_decay,
+            data_format=data_format,
+            enable_detail_summary=enable_detail_summary,
+        )
+
+        output_3x3 = conv_bn_act(
+            "conv_3x3",
+            output_1x1,
+            filters=growth_rate,
+            kernel_size=3,
+            weight_decay_rate=weight_decay_rate,
+            is_training=is_training,
+            activation=activation,
+            batch_norm_decay=batch_norm_decay,
+            data_format=data_format,
+            enable_detail_summary=enable_detail_summary,
+        )
+
+        if data_format == "NHWC":
+            concat_axis = -1
+        if data_format == "NCHW":
+            concat_axis = 1
+
+        output = tf.concat([inputs, output_3x3], axis=concat_axis)
+
+        if enable_detail_summary:
+            tf.summary.histogram('output', output)
+
+    return output
+
+
+def densenet(
+        name,
+        inputs,
+        num_layers,
+        growth_rate,
+        bottleneck_rate=4,
+        weight_decay_rate=0.0,
+        is_training=tf.constant(False),
+        activation=None,
+        batch_norm_decay=0.99,
+        data_format="NHWC",
+        enable_detail_summary=False,
+):
+    """Block of Densent.
+
+    paper: https://arxiv.org/abs/1608.06993
+    In the original paper, dense block consists from the layers
+    which batch norm -> activation(relu) -> convolution(1x1) and batch norm -> activation -> convolution(3x3).
+    But in this method, the order of each layer change to convolution -> batch norm -> activation.
+
+    Args:
+        name (str): Block name, as scope name.
+        inputs (tf.Tensor): Inputs.
+        num_layers (int): Number of dense layer which consits 1x1 and 3x3 cov.
+        growth_rate (int): How many filters (out channel) to add each layer.
+        bottleneck_rate (int): The factor to be calculated bottle-neck 1x1 conv output channel.
+            `bottleneck_channel = growth_rate * bottleneck_rate`.
+            The default value `4` is from original paper.
+        weight_decay_rate (float): Number of L2 regularization be applied to convolution weights.
+        is_training (tf.constant): Flag if training or not.
+        activation (callable): Activation function.
+        batch_norm_decay (float): Batch norm decay rate.
+        enable_detail_summary (bool): Flag for summarize feature maps for each operation on tensorboard.
+        data_format (string):  Fromat for inputs data. NHWC or NCHW.
+    Returns:
+        tf.Tensor: Output of current  block.
+    """
+
+    with tf.variable_scope(name):
+        x = inputs
+        for i in range(0, num_layers):
```

Comment:
`0` of range is needed?

Reply:
Thanks, I will change

Reply:
Fixed in https://github.com/blue-oil/blueoil/pull/212/commits/1260d4f50a1d3850cedfdc6929cabce496f3f19e